ui/components/Drawers/TransformationsDrawer/index.tsx

```diff
@@ -47,29 +46,37 @@ function TransformationsDrawer({
 		let aiTransformation = [];
 		let basicTransformations = [];
 
-		Object.entries(plugins).map(([, plugin]) => {
-			plugin.operations.map((op) => {
-				if (plugin?.name !== "Basic" && !plugin.credentials?.required)
-					return (aiTransformation = [...aiTransformation, { plugin, op }]);
-				else if (plugin?.name === "Basic") {
-					return (basicTransformations = [
-						...basicTransformations,
-						{ plugin, op },
-					]);
-				}
-			});
+		Object.entries(plugins).forEach(([_, plugin]) => {
+			if (allowedPlugins?.[plugin.identifier] !== undefined) {
```

Comment:
```suggestion
			if (allowedPlugins?.[plugin.identifier]) {
```
No need to explicite check for `undefined`